sdks/java/core/src/main/java/com/google/cloud/dataflow/sdk/io/UnboundedSource.java

```diff
@@ -110,8 +110,21 @@ public boolean requiresDeduping() {
      * <p>For example, this could be sending acknowledgement requests to an external
      * data source such as Pub/Sub.
      *
-     * <p>This may be called from any thread, potentially at the same time as calls to the
-     * {@code UnboundedReader} that created it.
+     * <p>Note that:
```

Comment:
I'll move to the unbounded source commit. I added them based on my experience implementing that.